hal/mbed_sleep_manager.c

```diff
@@ -74,6 +73,7 @@ us_timestamp_t mbed_time_deepsleep(void)
 
 #ifdef MBED_SLEEP_TRACING_ENABLED
 
+extern void mbed_error_printf(const char *format, ...);
```

Comment:
I don't see any reason you can't get the real declaration from `mbed_interface.h`.

Reply:
Fixed